pymatgen/analysis/fragmenter.py

```diff
@@ -82,7 +82,7 @@ def __init__(
         if edges is None:
             self.mol_graph = MoleculeGraph.with_local_env_strategy(molecule, OpenBabelNN())
         else:
-            _edges = {(edge[0], edge[1]): None for edge in edges}
+            _edges: dict[tuple[int, int], dict | None] = {(edge[0], edge[1]): None for edge in edges}
```

Comment:
This looks like magic to me. Didn't expect declaring the a super set of its actual type would solve the problem.